todo-app/src/Components/todoList.js

```diff
@@ -0,0 +1,75 @@
+import React, { Component } from "react";
+
+import TodoItem from "./TodoItem";
+import "./styles/todoList.css";
+
+export default class TodoList extends Component {
+  state = {
+    toBeUpdatedItem: "",
+    toBeUpdatedDeadline: ""
+  };
+
+  handleCancelEdit = () => {
+    this.props.handleToggleEdit(null, true);
+  };
+
+  handleToggleEdit = id => {
+    const todoItem = this.props.todos.find(item => item.id === id);
+    this.setState({
+      toBeUpdatedDeadline: todoItem.deadline,
+      toBeUpdatedItem: todoItem.text
+    });
+    this.props.handleToggleEdit(id);
+  };
+
+  handleUpdate = id => {
+    const { toBeUpdatedItem, toBeUpdatedDeadline } = this.state;
+    this.props.handleUpdate(id, toBeUpdatedItem, toBeUpdatedDeadline);
+  };
+
+  handleChangeItem = event => {
+    this.setState({ toBeUpdatedItem: event.target.value });
+  };
+
+  handleChangeDeadline = event => {
+    this.setState({ toBeUpdatedDeadline: event.target.value });
+  };
+
+  handleToggleDone = id => {
+    this.props.handleToggleDone(id);
+  };
+
+  renderAllTodos = () => {
+    const todoItems = this.props.todos;
+    return todoItems.map(item => {
+      return (
+        <TodoItem
+          key={item.id}
+          data={item}
+          handleToggleDone={this.handleToggleDone}
+          handleRemove={this.props.handleRemove}
+          isEditing={this.props.isEditing === item.id}
+          handleCancelEdit={this.handleCancelEdit}
+          handleToggleEdit={this.handleToggleEdit}
+          handleUpdate={this.handleUpdate}
+          toBeUpdatedItem={this.state.toBeUpdatedItem}
```

Comment:
You've moved all the state in `TodoList`, where each `TodoItem` has its own little form. I would suggest keeping the form state there, and using a callback only when the editing is complete. That way, you reduce a lot of your code!